feat(docs): improve separation of concern on analytics lib

alejoamiras opened the pull request:
This PR covers a couple of things:

- Separating the concerns of the analytics service from the NPS stuff (this will allow me to better copy everything to other docs)
- Adds local development with netlify (allows for testing the email subscription). By the looks of it (we already had the dependency) there was a point in time where this was in place.
- Correctly parses the "already subscribed" so we don't show it as an error. 
- Updates the README accordingly 

On tracking:
- Removes fallback tracking (it's super convoluted and unnecessary at this point)
- Renames the `MATOMO_ENV` to `ENV` since... it's actually the `ENV` 😬 
- Adds a goal that is submitting an NPS score
- Fixed "Custom Variables" since they are now called Dimensions
=> Tested and finally working! 😬 


On matomo:
- I've also set up the goals and customVariables on all our environments (dev, staging and prod) so we can now correctly test on different environments without breaking prod data.

You can use `foceNPS()` to show the NPS Widget.

<img width="1236" height="274" alt="image" src="https://github.com/user-attachments/assets/fe6ec9bb-3e88-4cf4-ad34-e48f459b6f9e" />


_(already fixed the revenue = 10 thing)_

Rumata888 commented on docs/.env.template:
```diff
@@ -0,0 +1,2 @@
+BREVO_API_KEY=your_api_key_here
+ENV=dev
```
Did you intend to add this?

alejoamiras replied:
Hey @Rumata888 yeap :) -- since it's the template people can copy and paste.

critesjosh reviewed (approved):
i didnt test it, but looks good

alejoamiras committed:
feat(docs): improve separation of concern on analytics lib

This PR covers a couple of things:

- Separating the concerns of the analytics service from the NPS stuff (this will allow me to better copy everything to other docs)
- Adds local development with netlify (allows for testing the email subscription). By the looks of it (we already had the dependency) there was a point in time where this was in place.
- Correctly parses the "already subscribed" so we don't show it as an error.
- Updates the README accordingly

On tracking:
- Removes fallback tracking (it's super convoluted and unnecessary at this point)
- Renames the `MATOMO_ENV` to `ENV` since... it's actually the `ENV` 😬
- Adds a goal that is submitting an NPS score
- Fixed "Custom Variables" since they are now called Dimensions
=> Tested and finally working! 😬

On matomo:
- I've also set up the goals and customVariables on all our environments (dev, staging and prod) so we can now correctly test on different environments without breaking prod data.

You can use `foceNPS()` to show the NPS Widget.

<img width="1236" height="274" alt="image" src="https://github.com/user-attachments/assets/fe6ec9bb-3e88-4cf4-ad34-e48f459b6f9e" />

_(already fixed the revenue = 10 thing)_

Co-authored-by: Alejo Amiras <alejo.amiras@gmail.com>
